litgpt/utils.py

```diff
@@ -557,13 +558,40 @@ def instantiate_bnb_optimizer(optimizer, model_parameters):
 
 
 def instantiate_torch_optimizer(optimizer, model_parameters, **kwargs):
+    # Special care taken where some optimizers do not have parameter "fused" like:
+    #   bnb.optim.AdamW8bit
+    #   grokadamw.GrokAdamW
+    
     if isinstance(optimizer, str):
-        optimizer_cls = getattr(torch.optim, optimizer)
+        if "." in optimizer:
+            class_module, class_name = optimizer.rsplit(".", 1)
+        else:
+            class_module, class_name = "torch.optim", optimizer
+
+        module = __import__(class_module, fromlist=[class_name])
+        optimizer_cls = getattr(module, class_name)
+
+        if "fused" in kwargs and "fused" not in inspect.signature(optimizer_cls).parameters:
+            kwargs = dict(kwargs)   # copy
+            del kwargs["fused"]
```

Comment:
Thanks for the PR! I am thinking maybe we can make this even more general so that it would also remove other unsupported arguments if present, something like

```suggestion
        if "." in optimizer:
            class_module, class_name = optimizer.rsplit(".", 1)
        else:
            class_module, class_name = "torch.optim", optimizer

        module = __import__(class_module, fromlist=[class_name])
        optimizer_cls = getattr(module, class_name)

        valid_params = set(inspect.signature(optimizer_cls).parameters)
        kwargs = {key: value for key, value in kwargs.items() if key in valid_params}
        optimizer = optimizer_cls(model_parameters, **kwargs)
```